core/src/main/scala/design/hamu/core/command/DecimalToBinaryCommand.scala

```diff
@@ -0,0 +1,28 @@
+package design.hamu.core.command
+
+object DecimalToBinaryCommand extends Command {
+
+  def run(ls: List[String]): String = ls match {
+    case decimal :: tail =>
+      var isNegative = false
+      val abs = (decimal: Int) => {
+        if (decimal >= 0) {
+          decimal
+        } else {
+          isNegative = true
+          -decimal
+        }
+      }
+      var mutableDecimal = abs(decimal.toInt)
+      var mutableString = ""
+      if (mutableDecimal == 0) {
+        s"0"
+      } else {
+        while (mutableDecimal != 0) {
```

Comment:
So in FP we don't really use `loops` because that usually means side effects and mutability like what we are doing with `mutableDecimal` and `mutableString`. You can accomplish the same effect using recursion.

You can try doing it yourself for practice but I created a gist that is fully functional here:
https://gist.github.com/vangogh500/e4b18e2e18629e244381193e24d835c0